flang/include/flang/Optimizer/Dialect/FIRType.h

```diff
@@ -372,6 +372,9 @@ bool isBoxedRecordType(mlir::Type ty);
 /// Return true iff `ty` is a type that contains descriptor information.
 bool isTypeWithDescriptor(mlir::Type ty);
 
+/// Return true if CLASS(*)
+bool isClassStarType(mlir::Type ty);
```

Comment:
nit: please move the declaration near `isUnlimitedPolymorphicType` and `isAssumedType`.